pkg/accountmanager/poller.go

```diff
@@ -126,16 +126,16 @@ func (p *accountPoller) addOrUpdateSelector(selector *crdv1alpha1.CloudEntitySel
 			}
 		}
 	}
+}
 
-	// Store selector to filter cloud resources based on selector.
-	p.selector = selector
+// removeSelector removes all vm inventory specific to the selector.
+func (p *accountPoller) removeSelector(selectorNamespacedName *types.NamespacedName) {
```

Comment:
This function can be removed now. Earlier it was resetting selector, now we dont store selector in the poller. May be remove the VM inventory from the caller.